File: intel-hex-to-bin.java/src/test/java/demo/RowTest.java
package demo;

import org.testng.annotations.Test;

import static org.testng.Assert.*;

/**
 * Unit tests for {@link Row}
 */
public class RowTest {

    @Test
    public void constructorTest() {
        Row row = new Row(":021234005678FF");

        assertFalse(row.isEmpty());
        assertEquals(row.getUnsigned8Bit(), (short) 0x02);
        assertFalse(row.isEmpty());
        assertEquals(row.getUnsigned16Bit(), (short) 0x1234);
        assertFalse(row.isEmpty());
        assertEquals(row.getUnsigned8Bit(), (short) 0x00);
        assertFalse(row.isEmpty());
        assertEquals(row.getUnsigned8Bit(), (short) 0x56);
        assertFalse(row.isEmpty());
        assertEquals(row.getUnsigned8Bit(), (short) 0x78);
        assertFalse(row.isEmpty());
        assertEquals(row.getUnsigned8Bit(), (short) 0xFF);
        assertTrue(row.isEmpty());
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void noColonConstructorTest() {
        new Row("00000001FF");
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void wrongCharsConstructorTest() {
        new Row(":XX");
    }

    @Test(expectedExceptions = IllegalArgumentException.class)
    public void oddCharsNumberConstructorTest() {
        new Row(":0");
    }

    @Test
    public void getUnsigned8BitTest() {
        Row row = new Row(":01ab");

        assertFalse(row.isEmpty());
        assertEquals(row.getUnsigned8Bit(), (short) 0x01);
        assertEquals(row.getUnsigned8Bit(), (short) 0xAB);
        assertTrue(row.isEmpty());
    }

    @Test
    public void getUnsigned16BitTest() {
        Row row = new Row(":1234abcd");

        assertFalse(row.isEmpty());
        assertEquals(row.getUnsigned16Bit(), 0x1234);
        assertEquals(row.getUnsigned16Bit(), 0xABCD);
        assertTrue(row.isEmpty());
    }

}

File: intel-hex-to-bin.kotlin/src/test/kotlin/demo/RowTest.kt
package demo

import org.testng.Assert.*
import org.testng.annotations.Test

/**
 * Unit tests for [Row]
 */
class RowTest {

    @Test
    fun constructorTest() {
        val row = Row(":021234005678FF")

        assertFalse(row.isEmpty())
        assertEquals(row.getUnsigned8Bit(), 0x02.toShort())
        assertFalse(row.isEmpty())
        assertEquals(row.getUnsigned16Bit(), 0x1234.toInt())
        assertFalse(row.isEmpty())
        assertEquals(row.getUnsigned8Bit(), 0x00.toShort())
        assertFalse(row.isEmpty())
        assertEquals(row.getUnsigned8Bit(), 0x56.toShort())
        assertFalse(row.isEmpty())
        assertEquals(row.getUnsigned8Bit(), 0x78.toShort())
        assertFalse(row.isEmpty())
        assertEquals(row.getUnsigned8Bit(), 0xFF.toShort())
        assertTrue(row.isEmpty())
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun noColonConstructorTest() {
        Row("00000001FF")
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun wrongCharsConstructorTest() {
        Row(":XX")
    }

    @Test(expectedExceptions = arrayOf(IllegalArgumentException::class))
    fun oddCharsNumberConstructorTest() {
        Row(":0")
    }

    @Test
    fun getUnsigned8BitTest() {
        val row = Row(":01ab")

        assertFalse(row.isEmpty())
        assertEquals(row.getUnsigned8Bit(), 0x01.toShort())
        assertEquals(row.getUnsigned8Bit(), 0xAB.toShort())
        assertTrue(row.isEmpty())
    }

    @Test
    fun getUnsigned16BitTest() {
        val row = Row(":1234abcd")

        assertFalse(row.isEmpty())
        assertEquals(row.getUnsigned16Bit(), 0x1234)
        assertEquals(row.getUnsigned16Bit(), 0xABCD)
        assertTrue(row.isEmpty())
    }

}
